airflow/jobs.py

```diff
@@ -1906,12 +1906,15 @@ def __init__(
         :param args:
         :param kwargs:
         """
+        if donot_pickle is None:
```

Comment:
This method is only invoked from here: https://github.com/apache/airflow/blob/master/airflow/models/__init__.py#L3960

Here the `donot_pickle` is also loaded from the config: https://github.com/apache/airflow/blob/master/airflow/models/__init__.py#L3933

Apart from that, it feels like we should set this on an Airflow level, instead of a job level.